scripts/fhir_scripts.sh

```diff
@@ -204,6 +204,108 @@ function igtools_version() {
     fi
 }
 
+function check_pytool_version() {
+    local package_name="$1"
+    local repo_url="$2"
+    local current_version=""
+    local latest_version=""
+    local response=""
+
+    # Handle missing parameters gracefully
+    if [[ -z "$package_name" || -z "$repo_url" ]]; then
+        echo "error:missing_parameters"
+        return 0
+    fi
+
+    # Get current version from pipx
+    current_version=$(sudo pipx list --global --short 2>/dev/null | awk -v pkg="$package_name" '$1 == pkg {print $2}') || true
+
+    if [[ -z "$current_version" ]]; then
+        echo "not_installed"
+        return 0
+    fi
+
+    # Try to fetch latest version from GitHub with 1s timeout
+    response=$(curl -s --connect-timeout 1 --max-time 1 "$repo_url" 2>/dev/null || true)
+
+    if [[ -z "$response" ]]; then
+        echo "error:fetch_failed_or_timeout:${current_version}"
+        return 0
+    fi
+
+    # Try to find versions from filenames (dist/*.tar.gz, *.whl)
+    latest_version=$(echo "$response" \
+        | grep -Eo "${package_name}-[0-9]+(\.[0-9]+){1,2}(\.tar\.gz|\.whl)" \
+        | sort -V \
+        | tail -n 1 \
+        | grep -Eo '[0-9]+(\.[0-9]+){1,2}' || true)
+
+    # If not found, try to extract from __VERSION__ in source code listing
+    if [[ -z "$latest_version" ]]; then
+        latest_version=$(echo "$response" \
+            | grep -Eo "__VERSION__\s*=\s*(Version\()?['\"][0-9]+(\.[0-9]+){1,2}['\"]\)?" \
+            | sed -E "s/.*__VERSION__\s*=\s*(Version\()?['\"]([^'\"]+)['\"]\)?/\2/" \
+            | sort -V \
+            | tail -n 1 || true)
+    fi
+
+    if [[ -z "$latest_version" ]]; then
+        echo "error:could_not_parse_latest_version:${current_version}"
+        return 0
+    fi
+
+    # Compare installed vs. latest
+    if [[ "$current_version" == "$latest_version" ]]; then
```

Comment:
Could you change that to a non-bash-specific comparison for more robustness?
```suggestion
    if [ "$current_version" = "$latest_version" ]; then
```